### ISSUE TITLE
[Bug] Error: Call to undefined method OCA\NCDownloader\Tools\Helper::getSearchSites()

### ISSUE THREAD

**Issue by TtuxX** [shiningw/ncdownloader#39]
Hi @shiningw ! :)

I hope you are doing fine.

I cannot use the latest version of NC Downloader anymore. 
Everytime I try to load the app, I get the following Internal Server Error:

    [index] Error: Call to undefined method OCA\NCDownloader\Tools\Helper::getSearchSites() in file '/usr/local/www/nextcloud/apps/ncdownloader/lib/Controller/MainController.php' line 79

    /usr/local/www/nextcloud/lib/private/AppFramework/App.php - line 157:
    OC\AppFramework\Http\Dispatcher->dispatch(OCA\NCDownlo ... "}, "Index")

    /usr/local/www/nextcloud/lib/private/Route/Router.php - line 302:
    OC\AppFramework\App::main("MainController", "Index", OC\AppFramew ... {}, { _route: "n ... "})

    /usr/local/www/nextcloud/lib/base.php - line 1006:
    OC\Route\Router->match("/apps/ncdownloader/")

    /usr/local/www/nextcloud/index.php - line 36:
    OC::handleRequest()

    Caused by Error: Call to undefined method OCA\NCDownloader\Tools\Helper::getSearchSites()

    /usr/local/www/nextcloud/apps/ncdownloader/lib/Controller/MainController.php - line 57:
    OCA\NCDownloader\Controller\MainController->buildParams()

    /usr/local/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php - line 217:
    OCA\NCDownloader\Controller\MainController->Index()

    /usr/local/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php - line 126:
    OC\AppFramework\Http\Dispatcher->executeController(OCA\NCDownlo ... "}, "Index")

    /usr/local/www/nextcloud/lib/private/AppFramework/App.php - line 157:
    OC\AppFramework\Http\Dispatcher->dispatch(OCA\NCDownlo ... "}, "Index")

    /usr/local/www/nextcloud/lib/private/Route/Router.php - line 302:
    OC\AppFramework\App::main("MainController", "Index", OC\AppFramew ... {}, { _route: "n ... "})

    /usr/local/www/nextcloud/lib/base.php - line 1006:
    OC\Route\Router->match("/apps/ncdownloader/")

    /usr/local/www/nextcloud/index.php - line 36:
    OC::handleRequest()

**Comment by shiningw** [comment]
Hi,

This could be caused by a partial upgrade. You should remove ncdownloader(also remove the directory) and try to install it again.Hope it helps you

**Comment by TtuxX** [comment]
Hi ! 

Thank you for this advice, however I can confirm this issue remains persistent:

1. I uninstalled and deleted the app version 0.8.6 from Nextcloud apps page. The ncdownloader folder is then deleted by Nextcloud.
2. I tried downloading and activating the latest app version 0.8.6 again from Nextcloud and the eror was the same.

Then I tried to install an older version to check:

1. I removed the 0.8.6 app again
2. Downloaded, extracted and activated NCDownloader version 0.8.0 => this version works without any issue on my instance
3. I then upgraded the app to version 0.8.6 from within Nextcloud 
4. The error appears again for version 0.8.6 (cannot launch the main NCDownloader page ; however the admin settings page for example works without any issue).
